feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -903,6 +959,53 @@ fun CaptureButton(
     }
 }
 
+/**
+ * Handler for using certain key events buttons as capture buttons.
+ */
+@Composable
+private fun CaptureKeyHandler(
+    onPress: (CaptureSource) -> Unit,
+    onRelease: (CaptureSource) -> Unit
+) {
+    val context = LocalContext.current
+    val view = LocalView.current
+
+    fun keyCodeToCaptureSource(keyCode: Int): CaptureSource = when (keyCode) {
+        KeyEvent.KEYCODE_VOLUME_UP -> CaptureSource.VOLUME_UP
+        KeyEvent.KEYCODE_VOLUME_DOWN -> CaptureSource.VOLUME_DOWN
+        else -> TODO("Keycode not assigned to CaptureSource")
+    }
+
+    DisposableEffect(context) {
+        val keyEventDispatcher = ViewCompat.OnUnhandledKeyEventListenerCompat { _, event ->
+            when (event.keyCode) {
+                KeyEvent.KEYCODE_VOLUME_UP, KeyEvent.KEYCODE_VOLUME_DOWN -> {
+                    val captureSource = keyCodeToCaptureSource(event.keyCode)
+                    if (event.action == KeyEvent.ACTION_DOWN) {
```

Comment:
Since volume buttons are meant to turn up/down volume, the "ACTION_DOWN" event will be called continually until you release. We probably only need to call `onPress()` once until it's released. You can keep a boolean `var` in this `DisposableEffect` to check if `ACTION_DOWN` has been called already, and set it to `false` when `ACTION_UP` is called.